### ISSUE TITLE
Selenium + IE11 + Outlook web: IEDriverServer crashed after click on Setting Button from Outlook web

### ISSUE THREAD

**Issue by oprilepin** [SeleniumHQ/selenium#6621]
## Meta -
OS: Windows 10 
Browser: Internet Explorer 11
C#, NUnit, Selenium WebDriver

## Actual Behavior - After click on Setting button - IEDriverServer is stopped working and no any other actions can be executed
I've executed my script from VS by NUnit (script is implemented with C# and Selenium WebDriver)
After Click on 'More mail Actions' for email IE is crashed and script is also failed with the following output:
## Output -
Message: System.Reflection.TargetInvocationException : Exception has been thrown by the target of an invocation.
  ----> OpenQA.Selenium.WebDriverException : The HTTP request to the remote WebDriver server for URL http://localhost:65204/session/1c48d735-6c80-435e-8faa-7dc6a4fe024b/element/4504e080-4c71-47d1-8390-3db745daef10/click timed out after 60 seconds.
  ----> System.Net.WebException : The request was aborted: The operation has timed out.

Notice: this output I've got after any actions for opened email in outlook ( reply, forward, settings etc)

## Steps to reproduce -
Go to https://outlook.office.com/owa/
Login as User
Open some email from Inbox folder
Click 'More mail Actions' button
Try to click 'Get Add-Ins' from sub menu

## The part of code -
Controls defined:
        // More Mail Actions Button
        [FindsBy(How = How.XPath, Using = "//button[contains(@aria-label, 'More mail actions') and 
         @role='menuItem']")]
        public IWebElement MoreMailActionsButton { get; set; }

        // Get Add-Ins Button
        [FindsBy(How = How.XPath, Using = "//span[text()='Get Add-ins']")]
        public IWebElement GetAddInsButton { get; set; }

Actions:
MoreMailActionsButton.Click();
GetAddInsButton.Click();

**Comment by barancev** [comment]
For issues please provide a [concise reproducible test case](http://sscce.org/) and describe what results you are seeing and what results you expect.

See [CONTRIBUTING.md](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#issue-contributions)

**Comment by oprilepin** [comment]
> For issues please provide a [concise reproducible test case](http://sscce.org/) and describe what results you are seeing and what results you expect.
> 
> See [CONTRIBUTING.md](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#issue-contributions)

Hi Alex @barancev, I've updated my issue request as I see that from my implementation

**Comment by oprilepin** [comment]
Hi @barancev looks I've found where the original problem is, so, it's driver.Quit() / driver.Close() / driver.Dispose() methods - any from these doesnt work. I mean in TearDown after tests, driver is not killed and browser also is not closed. Could you please help me here? Thx

**Comment by barancev** [comment]
@oprilepin How driver.Quit() / driver.Close() / driver.Dispose() methods invoked after tests can affect Click operation?

**Comment by oprilepin** [comment]
@barancev driver.Quit() / driver.Close() / driver.Dispose() methods invoked after tests don't affect Click method. Here I mean that after Click(), driver.Quit() / driver.Close() / driver.Dispose() anyone fro these doesn't work. Please see two examples below:

PASSED:
IJavaScriptExecutor executor = (IJavaScriptExecutor)driver;
executor.ExecuteScript("arguments[0].click();", element1);

 [TearDown] 
 public void TearDownTest()
 {
     driver.Quit();
 }

-----------------------------------------------------------------------------

FAILED:

element1.Click();
 [TearDown] 
 public void TearDownTest()
 {
     driver.Quit();
 }

NOTICE: Click action is completely done - but driver.Quit() works only after IJSExecutor but not after default IWebDriver Click method

**Comment by jimevans** [comment]
@oprelepin What version of Windows 10 are you using?

**Comment by oprilepin** [comment]
> @oprelepin What version of Windows 10 are you using?

Windows 10 Enterprise

**Comment by jimevans** [comment]
@oprilepin Not the edition (“Professional”, “Enterprise”, etc.), but the actual version number, please. For example, the Windows 10 October 2018 Update shipped with a bug in IE that causes crashes on close. In that case, it’s a browser bug, not a driver bug, and there’s nothing the Selenium project can do about it.

**Comment by oprilepin** [comment]
@jimevans What is the solution for this case if I have latest Window updates that is causing IE bug?

**Comment by jimevans** [comment]
@oprilepin Wait for the hotfix to be released by Microsoft, which will go out in a “Patch Tuesday” release soon. The issue was reported to Microsoft, and they have fixed it, but it was reported too late to be included with the Windows 10 October 2018 Update. I had hoped it would make it into the October patch update in late October, but I think it didn’t make that either. It should be included sometime here in November. Contact Microsoft for more concrete details.

**Comment by oprilepin** [comment]
@jimevans thanks for help and explanation in case of this. But, I've also analyzed my script and situation where am I and what I've found:

PASSED:
element1.SendKeys('123');
driver.Quit();

FAILED:
element1.SendKeys('123');
element2.Click();
driver.Quit();

PASSED:
element1.SendKeys('123');
IJavaScriptExecutor executor = (IJavaScriptExecutor)driver;
executor.ExecuteScript("arguments[0].click();", element2);
driver.Quit();

It means that after default WebDriver CLick() method, something went wrong and Quit() OR something between Click() <-> Quit() crashes.
But when I complete my script with JS Executor Click(), then Quit() works as expected

I'm not fully sure how it happens, but yes that case I can see every time and should use IJS Click + default SendKeys()

**Comment by jimevans** [comment]
The specific bug in Windows 10/IE is directly related to the browser’s receiving of mouse movement messages, then attempting to close the window. When running with the default IE driver configuration, the driver simulates mouse movement messages during a call to `Click()`. So yes, it’s directly related to `Click()`, then `Quit()`. This is absolutely not a bug that the driver nor the language bindings can do anything about.

How do I know this? I’m the one who found and reported the issue to Microsoft. I examined the stack traces into when the crash occurs using the symbol files for Windows provided by the Windows SDK. I have tested the fixed version of IE, both while the fix was being developed, and after it was released into Windows Insider builds.

I can state with complete confidence that there is an issue. It’s in the browser itself. It will require a new version of the browser from the browser vendor (Microsoft) to resolve it. I have less confidence that the issue you’re having is the issue I reported, but the symptoms are identical, and until the fix is released, there is no way to know with confidence whether the issue you’re experiencing is the same one or a different one altogether.
